File: src/StoreKeeper.cpp
#include "StoreKeeper.hpp"
#include "Supply.hpp"
#include "Warehouse.hpp"
#include <iostream>
using namespace std;
StoreKeeper::StoreKeeper() : Employee() {}

StoreKeeper::StoreKeeper(const string& name, const string& password, const string& role, int age, const string& post, int salary) : Employee(name, password, role, age, post, salary) {}

bool StoreKeeper::acceptShipment(Shipment* shipment, Warehouse* warehouse) {
    cout << "Кладовщик " << getName() << " принимает отгрузку" << endl;
    if (shipment && warehouse) {
        bool ok = warehouse->processShipment(shipment);
        cout << (ok ? "Отгрузка успешно обработана" : "Ошибка при обработке отгрузки") << endl;
        return ok;
    }
    return false;
}

File: warehouse/StoreKeeper.java
package com.warehouse;

public class StoreKeeper extends UserSystem {
    private int employeeId;
    private static int nextEmployeeId = 1;
    
    public StoreKeeper(String name, String role, String password) {
        super(name, role, password);
        this.employeeId = nextEmployeeId++;
    }
    
    public boolean acceptSupply(Supply supply, Warehouse warehouse) {
        System.out.println("Кладовщик " + getName() + " принимает поставку #" + supply.getSupplyId());
        if (supply != null && warehouse != null) {
            supply.processSupply(warehouse);
            return true;
        }
        return false;
    }
    
    public void displayInfo() {
        System.out.println("Кладовщик: " + getName() + ", ID: " + employeeId);
    }
    
    public int getEmployeeId() { return employeeId; }
}